src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -80,4 +74,60 @@ public GHClient(IProgram program)
         {
         }
     }
+
+    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
+    [ProvideService(typeof(IUIProvider), IsAsyncQueryable = true)]
+    [ProvideAutoLoad(UIContextGuids.NoSolution)]
+    [ProvideAutoLoad(UIContextGuids.SolutionExists)]
+    [Guid("D5CE1488-DEDE-426D-9E5B-BFCCFBE33E53")]
+    public sealed class ServiceProviderPackage : AsyncPackage
+    {
+        static Version vsversion;
+        public static Version VSVersion
+        {
+            get
+            {
+                if (vsversion == null)
+                {
+                    var asm = typeof(ITaskList).Assembly;
+                    try
+                    {
+                        // this will return Microsoft.VisualStudio.Shell.Immutable.14.0 in VS15
+                        // but Microsoft.VisualStudio.Shell.Framework in Dev15
+                        var vinfo = FileVersionInfo.GetVersionInfo(asm.Location);
+                        vsversion = new Version(vinfo.FileMajorPart, vinfo.FileMinorPart, vinfo.FileBuildPart, vinfo.FilePrivatePart);
+                    }
+                    catch
+                    {
+                        // something wrong, fallback to assembly version
+                        vsversion = asm.GetName().Version;
+                    }
+                }
+                return vsversion;
+            }
+        }
+
+        protected override async tasks.Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
+        {
+            AddService(typeof(IUIProvider), CreateService, true);
+
+            // Load the start page package only for Dev15 Preview 4
+            if (VSVersion.Major == 15 && VSVersion.Build == 25618)
+            {
+                var shell = await GetServiceAsync(typeof(SVsShell)) as IVsShell;
+                IVsPackage startPagePackage;
+                if (ErrorHandler.Failed(shell?.LoadPackage(new Guid("3b764d23-faf7-486f-94c7-b3accc44a70d"), out startPagePackage) ?? -1))
+                {
+                    // ¯\_(ツ)_/¯
```

Comment:
Can we display a full screen vector art version of this ascii drawing in this case and play the sad trombone music?

:trollface: